src/Razor/src/Microsoft.VisualStudioCode.RazorExtension/Services/LspDynamicFileProvider.LspTextChangesTextLoader.cs

```diff
@@ -45,11 +45,22 @@ public override async Task<TextAndVersion> LoadTextAndVersionAsync(LoadTextOptio
             var sourceText = await _document.GetTextAsync(cancellationToken).ConfigureAwait(false);
 
             // Validate the checksum information so the edits are known to be correct
-            if (IsSourceTextMatching(sourceText))
+            try
             {
-                var version = await _document.GetTextVersionAsync(cancellationToken).ConfigureAwait(false);
-                var newText = ApplyChanges(sourceText, _changes);
-                return TextAndVersion.Create(newText, version.GetNewerVersion());
+                if (IsSourceTextMatching(sourceText))
+                {
+                    var version = await _document.GetTextVersionAsync(cancellationToken).ConfigureAwait(false);
+                    var newText = ApplyChanges(sourceText, _changes);
+                    return TextAndVersion.Create(newText, version.GetNewerVersion());
+                }
+            }
+            catch (ArgumentException ex)
```

Comment:
Based on https://prism.vsdata.io/failure/?query=%20sev!%3DDiagnostic%20p%3Dcsdevkit&eventType=fault&failureType=hits&failureHash=a111c29f-cca2-7bfd-4add-e5573c3797ee

I left it as isolated as possible but should we just catch all exceptions? We would still want to report a fault (needs follow up with EA changes) but it can be handled

Reply:
Yeah, IMO we should catch all exceptions except `OperationCancellationException`. We should let that bubble up to the caller on the Roslyn side.